src/orchestrator/src/lib.rs

```diff
@@ -137,3 +122,25 @@ impl CpuLimit {
         self.millicpus
     }
 }
+
+impl<'de> Deserialize<'de> for CpuLimit {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        // Note -- we just round off any precision beyond 0.001 here.
+        let float = f64::deserialize(deserializer)?;
+        let millicpus = (float * 1000.).round();
+        if millicpus < 0. || millicpus > (std::usize::MAX as f64) {
+            use serde::de::Error;
+            Err(D::Error::invalid_value(
+                Unexpected::Float(float),
+                &"a float representing a plausible number of CPUs",
+            ))
+        } else {
+            Ok(Self {
+                millicpus: millicpus as usize,
+            })
+        }
+    }
+}
```

Comment:
Should we just make it take a float directly?

Reply:
I think that would be a bit of an abstraction violation. If we've decided that thousandths of a CPU (or whatever) is our unit of compute, then logically it is an integral type, and the only reason we have floats in the pipeline is just an artifact of the fact that JSON doesn't support fixed-point decimals. So I wanted these to live as floats for as short a time as possible.

TBH I am just prejudiced against floats! I think they belong in numerical simulations and anywhere else they're used is a bit of an antipattern.

Reply:
I agree with the prejudice against floats! That's why I made these `usize` to begin with, I think. Let me ask the reverse question, then: what if we just declare that the units are `millicpus` in the JSON, and require that you show up with an integer? Then you can just derive `Deserialize` normally, no?

Reply:
Can rename the field name `millicpu_limit` to make it clear.

Reply:
eh, if you feel strongly, we can do that, but personally I think being able to type cpus: 2.5 is nicer than mcpus: 2500 or whatever.

Reply:
It's single-user though, right? Like we're expecting that there is only ever one place in Materialize Cloud where this gets set? No strong feels, but feels like a moment to optimize for writing the least code rather than having a human-friendly UX. (Hm, I guess this argument applies to using `ByteSize` vs a field named `memory_limit_bytes`, too. 🤷🏽‍♂️)